bindings/experimental/distrdf/python/DistRDF/HeadNode.py

```diff
@@ -528,24 +467,21 @@ def build_rdf_from_range(current_range: Ranges.TreeRangePerc) -> TaskObjects:
             The function creates a TChain from the information contained in the
             input range object. If the chain cannot be built, returns None.
             """
-            # Prepare main TChain
-            chain, entries_in_trees = build_chain_from_range(current_range)
-            if chain is None:
-                # The chain could not be built.
+
+            clustered_range, entries_in_trees = Ranges.get_clustered_range_from_percs(current_range)
+
+            if clustered_range is None:
                 return TaskObjects(None, entries_in_trees)
 
-            # Create RDataFrame object for this task
-            if defaultbranches is not None:
-                rdf = ROOT.RDataFrame(chain, defaultbranches)
-            else:
-                rdf = ROOT.RDataFrame(chain)
+            treeandfilenameglobs = [
+                (treename, filename) for treename, filename in zip(clustered_range.treenames, clustered_range.filenames)
+            ]
```

Comment:
can this be just

```python
treeandfilenameglobs = zip(clustered_range.treenames, clustered_range.filenames)
```
?

Reply:
I wonder if cppyy can convert the zip iterator to an `std::vector`, I will try

Reply:
Interestingly, it seems to work. I will modify the code accordingly